WordPress/gutenberg
# Don't send empty paragraph tags to view on Android

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1970

Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2307

### Description
This change is for fixing missing placeholder text on Quote and Pullquote blocks on Android. The updates here detect in `RichText` on Android when we are sending empty paragraph tags to `RCTAztecView` and send an empty string instead.

We then add a check to onTextUpdate to ensure that empty paragraph text does not get passed back to the RichText Value without having its paragraph tags added back. 

### How has this been tested?




1. Add Quote and Pullquote blocks, verify that the "Enter Quote" placeholder text is visible.
2. Try adding multiline quotes and merging other blocks into a quote block, make sure no text is lost.
3. Smoke test other rich text based blocks to ensure there is no unintended behavior
4. Add empty, and multiline quotes and make sure the html shows paragraph tags for each line of text. 

*Note that this fix does not resolve the extra backspace issue seen in other blocks (https://github.com/wordpress-mobile/gutenberg-mobile/issues/1524). The simple case of typing a quote and then pressing backspace seems to reshow the placeholder correctly, but if you press enter to make a multiline quote, and then backspace til empty, the placeholder will not show similar to how we need an extra backspace in other blocks.

### Screenshots 
<img width="446" alt="Screen Shot 2020-05-22 at 6 20 29 PM" src="https://user-images.githubusercontent.com/1103838/82718711-f7ff9a00-9c58-11ea-8d8e-a9707e11664e.png">

### Checklist:
